classes/migration/upgrade/v3_4_0/PreflightCheckMigration.php

```diff
@@ -550,6 +551,31 @@ public function up(): void
                         throw new \Exception('A row with setting_name="submissionChecklist" found in table ' . $this->getContextSettingsTable() . " without the expected setting_value. Expected an array encoded in JSON but found:\n\n" . $value . "\n\nFix or remove this row before upgrading.");
                     }
                 });
+
+            // Check if database engine supports foreign key constraints, see pkp/pkp-lib#6732
+            if (DB::connection() instanceof MySqlConnection) {
+                $defaultEngine = DB::scalar('SELECT ENGINE FROM INFORMATION_SCHEMA.ENGINES WHERE SUPPORT = "DEFAULT"');
+                if ($defaultEngine !== 'InnoDB') {
+                    throw new Exception(
+                        'A default database engine ' . $defaultEngine . ' isn\'t supported, expecting InnoDB. ' .
+                        'Please change the default database engine to InnoDB to run the upgrade.'
+                    );
+                }
+
+                $result = DB::select(
```

Comment:
Since we've already checked that the default engine is `InnoDB` above, this query doesn't need `information_schema.engines`. It could just be:
```sql
SELECT t.table_name, t.engine AS table_engine
FROM information_schema.tables AS t
WHERE t.table_schema = :databaseName AND t.engine <> 'InnoDB';
```
(I'm a little nervous we're going to run into issues with more restrictive grants out there in the wild, so trying to keep these as simple as possible.)

I'd also suggest making the exception list any tables that need attention, rather than just throwing a generic exception if any tables match. Since the query is providing the information anyway :)